Make the calendar configurable

rowanmanning opened the pull request:
You can now choose from a list of predefined calendars when creating a
campaign. For now this includes Harptos, Odreian, and Tide.

Odreian is the default at the moment as I don't want this PR to affect
existing campaigns. Because there aren't very many at the moment we
could just manually add the calendar property to each of them and then
set the default to Harptos.

I've added a version number to each of the calendar IDs so that we can
iterate on them later if we want to without old campaign dates being
knocked out.

Matt: I expect that there's loads wrong with this because I'm not at all
familiar with Meteor or React.

rowanmanning committed:
Replace odreian-date with dream-date

Currently this is just a replacement of odreian-date with dream-date,
but it gets us one step closer to completing #6. Next step is to allow
you to choose from a predefined list of calendars, I think being able to
set up your own calendar through almanac will come later on.

rowanmanning commented on client/blocks/quests/index.js:
```diff
@@ -48,7 +48,7 @@ const Objective = connectObjective(({
 
 	{objective.completed && <Completed>
-		{new OdreianDate(objective.completedDate).format`${'llll'}`}
+		{new OdreianDate(objective.completedDate).format`${'h'}:${'mm'}${'a'}, ${'dddd'}, ${'Do'} of ${'MM'}, ${'YY'}`}
```
Note: this is hardcoded for now until I tackle rowanmanning/dream-date#9

rowanmanning committed:
Make campaign calendars configurable

You can now choose from a list of predefined calendars when creating a
campaign. For now this includes Harptos, Odreian, and Tide.

Odreian is the default at the moment as I don't want this PR to affect
existing campaigns. Because there aren't very many at the moment we
could just manually add the calendar property to each of them and then
set the default to Harptos.

I've added a version number to each of the calendar IDs so that we can
iterate on them later if we want to without old campaign dates being
knocked out.

Matt: I expect that there's loads wrong with this because I'm not at all
familiar with Meteor or React.

apaleslimghost commented on client/pages/home.js:
```diff
@@ -29,6 +30,12 @@ export default withCampaignData(({campaigns, createCampaign}) => <ul>
 	<li>
 		<form onSubmit={createCampaign}>
 			<input placeholder='Campaign' name='title' />
+			<select name='calendar'>
```
eventually this will want to move to a campaign settings page but it's fine here for now

apaleslimghost reviewed (dismissed):
this is actually perfect. there's not a single thing i would have done differently 👌

rowanmanning committed:
Fix an issue with Harptos dates

The Harptos calendar has more than 12 months, so this was just failing
to display if the month was set to any after 12.

For now I've just made all later months use the 12th ornament.

rowanmanning committed:
Update dream-date and add a Harptos-Common calendar

People might prefer to use the Harptos common names for the months of
the year. I was adding it anyway and thought I should expose it here.

apaleslimghost commented on client/blocks/time/display.js:
```diff
@@ -32,7 +32,7 @@ const ornaments = [
 ];
 
-const OrnamentedMonth = ({date}) => <Ornamented ornament={ornaments[date.monthIndex]} large>
+const OrnamentedMonth = ({date}) => <Ornamented ornament={ornaments[Math.min(date.monthIndex, 11)]} large>
```
so, the ornaments were kind of chosen to match the odreian month names. maybe at some point we want to do configurable display styles for the time block?

apaleslimghost replied:
<img width="338" alt="screen shot 2018-04-20 at 21 54 01" src="https://user-images.githubusercontent.com/631757/39073374-7656dddc-44e5-11e8-9679-9730ee7007c8.png">

rowanmanning replied:
Yeah having them configurable would be nice. They'll still work the same way for calendars with 12-month years
